os/include/tinyara/preference.h

```diff
@@ -44,6 +44,7 @@ enum preference_type_e {
 	PREFERENCE_TYPE_DOUBLE,
 	PREFERENCE_TYPE_BOOL,
 	PREFERENCE_TYPE_STRING,
+	PREFERENCE_TYPE_BINARY,
```

Comment:
It looks it's used in the framework only. Do we need to maintain here, not in the framework include?

Reply:
I agree, the entire enumeration can be moved to framework specific include path.
However, it should be done in a separate commit.